aliases.go

```diff
@@ -78,3 +104,57 @@ func (client *Client) DeleteAlias(input AliasDeleteInput) error {
 	err := client.Mutate(&m, v, WithName("AliasDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileAliases manages aliases API operations for AliasOwnerInterface implementations
+//
+// Aliases not in 'aliasesWanted' will be deleted, new tags from 'aliasesWanted' will be created. Reconciled aliases are returned.
+func (client *Client) ReconcileAliases(resourceType AliasOwnerInterface, aliasesWanted []string) ([]string, error) {
+	var deleteAliasFunc func(string) error
+	var existingAliases []string
+	var resourceId ID
+
+	switch resource := resourceType.(type) {
```

Comment:
Can remove this switch statement just like the one in `ReconcileTags()` by 

1. Making the `AliasOwnerInterface` provide a `ResourceID()` function (returns an `ID` always so we don't have the case for InfrastructureResource)
2. Add an `Aliases()` function so we don't have to use this switch statement to get `ManagedAliases` vs `Aliases`

Makes for a better opslevel-go experience so that customers don't also have to write these types of `switch` statements.

Reply:
@taimoorgit Customers don't have to write these kinds of switch statements. This is a very specific case that's being addressed and there is no apparent need for an interface at all. These added functions add no apparent value outside this function

Reply:
This feels really awkward.  It feels optimized for having a single function on the client `ReconcileAliases` when the purpose of an interface is to have a single function on an object that you call to do the work.

I wonder if it would feel better and also be easier to use if the interface was placing the `ReconcileAliases` interface on the individual resource object and then its implemented per resource object (just like a standard interface would be) something the end up being used like this

```
domain, _ := client.GetDomain("my-domain")
domain.ReconcileAliases(client, []string{"one","three","four"})
```